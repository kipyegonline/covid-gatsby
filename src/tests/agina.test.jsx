import React, { useState } from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON, { toJson } from "enzyme-to-json";
import { MyForm, Countries, CountryList } from "./hoc";

Enzyme.configure({ adapter: new Adapter() });

describe("<MyForm/> issues", () => {
  let wrapper = shallow(<MyForm formTitle={"Angela Agina"} />);

  test("snapshot", () => expect(toJSON(wrapper)).toMatchSnapshot());
  test("para", () =>
    expect(wrapper.find("p").text()).toBe("You just want attention"));
  test("presence of a list", () => {
    expect(wrapper.find("li")).toHaveLength(3);
  });
  test("submit button", () => {
    const props = {
      type: "submit",
      value: "Send Naah",
      className: "btn blue lighten-1",
    };
    expect(wrapper.find("input[type='submit']").props()).toEqual(props);
  });
});

const TextArea = ({ value, thanks }) => {
  const [wrapper, setWrapper] = useState("");
  return (
    <div>
      <h3>{wrapper}</h3>
      <label>*Enter Message</label>
      <textarea
        onChange={(e) => setWrapper(e.target.value)}
        value={value}
      ></textarea>
      <p className="text-sucess">{thanks}</p>
    </div>
  );
};

test("<Text area/>", () => {
  const _change = jest.fn();
  const wrapper = shallow(
    <TextArea value={10} onChange={_change} thanks="Karibu" />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
  expect(wrapper).toHaveLength(1);
  expect(wrapper.find("textarea").prop("value")).toEqual(10);
  wrapper.setProps({ value: 20 });
  expect(wrapper.find("textarea").prop("value")).toEqual(20);
  expect(wrapper.find("p").text()).toEqual("Karibu");
  expect(wrapper.find("h3").text()).toEqual("");
  wrapper
    .find("textarea")
    .simulate("change", { target: { value: "Jules is raining" } });
  //wrapper.update();
  expect(wrapper.find("h3").text()).toEqual("Jules is raining");
});
