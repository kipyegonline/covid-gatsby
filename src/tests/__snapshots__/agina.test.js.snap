// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyForm/> issues snapshot 1`] = `
<form
  className="form p-5 mx-auto my-2 gray lighten-2"
  onSubmit={[Function]}
>
  <h3>
    Angela Agina
  </h3>
  <h4 />
  <p>
    You just want attention
  </p>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      key="1"
    >
      All you need to know
       
      <span
        className="text-danger ml-5"
        onClick={[Function]}
      >
        X
      </span>
    </li>
    <li
      className="list-group-item"
      key="2"
    >
      I need your love
       
      <span
        className="text-danger ml-5"
        onClick={[Function]}
      >
        X
      </span>
    </li>
    <li
      className="list-group-item"
      key="3"
    >
      Unit testing manenos
       
      <span
        className="text-danger ml-5"
        onClick={[Function]}
      >
        X
      </span>
    </li>
  </ul>
  <label>
    Enter text
  </label>
  <textarea
    className="form-control my-2"
    onChange={[Function]}
    value=""
  />
  <input
    className="btn blue lighten-1"
    type="submit"
    value="Send Naah"
  />
</form>
`;

exports[`<Text area/> 1`] = `
<div>
  <h3 />
  <label>
    *Enter Message
  </label>
  <textarea
    onChange={[Function]}
    value={10}
  />
  <p
    className="text-sucess"
  >
    Karibu
  </p>
</div>
`;
